### PR TITLE
EVA-3540  move recover state from constructor to generate accessions

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

File: accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/generators/monotonic/MonotonicAccessionGenerator.java [hunk 1]
```diff
@@ -105,20 +70,43 @@ private static void assertBlockParametersAreInitialized(ContiguousIdBlockService
         }
     }
 
+    private void recoverState() {
+        if (RUN_RECOVERY && monotonicDatabaseService != null) {
```
Comment (tcezard) [inline accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/generators/monotonic/MonotonicAccessionGenerator.java:74]: Should we check that the generator has not SHUTDOWN ?

**Review by tcezard — APPROVED** [review]
Change looks ok.

File: accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/generators/monotonic/MonotonicAccessionGenerator.java (hunk 1 above)
Comment (nitin-ebi) [reply accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/generators/monotonic/MonotonicAccessionGenerator.java:74]: This is a private method and the public method from which this is being called is already doing the check.

Comment (tcezard) [reply accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/generators/monotonic/MonotonicAccessionGenerator.java:74]: That makes sense.

**Pull request merged**